vendor/fknsrs.biz/p/ottoext/loop/loop.go

```diff
@@ -144,46 +144,22 @@ func (l *Loop) processTask(t Task) error {
 	return nil
 }
 
-// Run handles the task scheduling and finalisation. It will block until
-// there's no work left to do, or an error occurs.
+// Run handles the task scheduling and finalisation.
+// It runs infinitely waiting for new tasks.
 func (l *Loop) Run() error {
-	for {
-		l.lock.Lock()
-		if len(l.tasks) == 0 {
-			// prevent any more tasks entering the ready channel
-			l.closed = true
-
-			l.lock.Unlock()
-
-			break
-		}
-		l.lock.Unlock()
-
-		t := <-l.ready
-
-		if t != nil {
-			if err := l.processTask(t); err != nil {
-				return err
-			}
+	for t := range l.ready {
+		if t == nil {
+			continue
 		}
-	}
 
-	// drain ready channel of any existing tasks
-outer:
-	for {
-		select {
-		case t := <-l.ready:
-			if t != nil {
-				if err := l.processTask(t); err != nil {
-					return err
-				}
-			}
-		default:
-			break outer
+		err := l.processTask(t)
+		if err != nil {
+			// TODO(divan): do we need to report
+			// errors up to the caller?
+			// Ignoring for now, as loop
+			// should keep running.
+			continue
 		}
 	}
-
-	close(l.ready)
-
 	return nil
 }
```

Comment:
I don't think changing it here is a good idea. At least, this will be pain when we introduce a vendor library. Do you have rights to create new repositories under github.com/status-im? I think you should fork the initial repository to our org, fix it there and change the dependency in status-im.

Reply:
@tiabc agree. I was thinking to open an issue in the upstream and wait until it's fixed there. But forking will be reliable way to go anyway. Confirming that I do have rights to create new repos.